src/reducers.ts

```diff
@@ -51,3 +53,9 @@ function moveShape(state: State, action: MoveShapeAction) {
     })
   });
 }
+
+function bringShapeForward(state: State, action: BringShapeForwardAction) {
```

Comment:
My recommendation is to keep these helper functions generic. So, instead of passing in the state and action, you could instead do:

```ts
function bringShapeForward(shapes: Shape[], shapeIndex: number): Shape[] {
  return [...shapes, shapes[action.shapeIndex]];
}
```

This way you can reuse this function in other parts of your app, and not rely on the action or state.